File: internal/domain/friendship/friendship.go
package friendship

import (
	"time"

	"gorm.io/gorm"
)

// FriendshipStatus 好友关系状态
type FriendshipStatus string

const (
	FriendshipStatusPending   FriendshipStatus = "pending"   // 待处理
	FriendshipStatusAccepted  FriendshipStatus = "accepted"  // 已接受
	FriendshipStatusRejected  FriendshipStatus = "rejected"  // 已拒绝
	FriendshipStatusBlocked   FriendshipStatus = "blocked"   // 已拉黑
)

// Friendship 好友关系实体
type Friendship struct {
	ID          string           `gorm:"primaryKey;type:varchar(36)" json:"id"`
	RequesterID string           `gorm:"type:varchar(36);not null;uniqueIndex:idx_friendship_pair" json:"requester_id"`
	AddresseeID string           `gorm:"type:varchar(36);not null;uniqueIndex:idx_friendship_pair" json:"addressee_id"`
	Status      FriendshipStatus `gorm:"type:varchar(20);not null;default:'pending'" json:"status"`
	CreatedAt   time.Time        `json:"created_at"`
	UpdatedAt   time.Time        `json:"updated_at"`
	DeletedAt   gorm.DeletedAt   `gorm:"index" json:"-"`

	// 关联
	Requester   *UserBrief       `gorm:"-" json:"requester,omitempty"`
	Addressee   *UserBrief       `gorm:"-" json:"addressee,omitempty"`
}

// UserBrief 用户简要信息（用于关联展示）
type UserBrief struct {
	ID     string `json:"id"`
	Email  string `json:"email"`
	Name   string `json:"name"`
	Avatar string `json:"avatar,omitempty"`
	Status string `json:"status,omitempty"`
}

func (Friendship) TableName() string {
	return "friendships"
}

File: openAIMClient/Mac/openAIM/openAIM/Models/Friendship.swift
//
//  Friendship.swift
//  openAIM
//
//  Created by Claude on 2026/3/8.
//

import Foundation

/// 好友关系状态
enum FriendshipStatus: String, Codable {
    case pending = "pending"
    case accepted = "accepted"
    case rejected = "rejected"
    case blocked = "blocked"
}

/// 好友关系状态（从当前用户视角）
enum FriendshipStatusView: String, Codable {
    case none = "none"
    case pendingSent = "pending_sent"
    case pendingReceived = "pending_received"
    case accepted = "accepted"
    case blocked = "blocked"
    case blockedByMe = "blocked_by_me"
}

/// 好友关系
struct Friendship: Codable, Identifiable, Hashable {
    let id: String
    let requesterId: String?
    let addresseeId: String?
    let status: FriendshipStatus?
    let createdAt: Date
    let updatedAt: Date?

    // 关联用户信息
    var user: User?
    var requester: User?
}

/// 搜索用户结果
struct SearchUser: Codable, Identifiable, Hashable {
    let id: String
    let email: String
    let name: String?
    let avatar: String?
    let status: String?
    let online: Bool?
    var friendshipStatus: FriendshipStatusView?
}

/// 搜索用户响应
struct SearchUsersResponse: Codable {
    let total: Int
    let page: Int
    let pageSize: Int
    let items: [SearchUser]
}

/// 好友列表响应
struct FriendsListResponse: Codable {
    let total: Int
    let page: Int
    let pageSize: Int
    let items: [Friendship]
}

/// 发送好友请求
struct SendFriendRequest: Codable {
    let userId: String
}

/// 好友请求数量响应
struct FriendRequestsCountResponse: Codable {
    let count: Int
}